state/application.go

```diff
@@ -31,12 +31,39 @@ import (
 	"github.com/juju/juju/core/leadership"
 	"github.com/juju/juju/core/model"
 	"github.com/juju/juju/core/network"
+	"github.com/juju/juju/core/network/firewall"
 	"github.com/juju/juju/core/status"
 	mgoutils "github.com/juju/juju/mongo/utils"
 	stateerrors "github.com/juju/juju/state/errors"
 	"github.com/juju/juju/tools"
 )
 
+// ExposedEndpoint encapsulates the expose-related details of a particular
+// application endpoint with respect to the sources (CIDRs or space IDs) that
+// should be able to access the ports opened by the application charm for an
+// endpoint.
+type ExposedEndpoint struct {
+	// A list of spaces that should be able to reach the opened ports
+	// for an exposed application's endpoint.
+	ExposeToSpaceIDs []string `bson:"to-space-ids,omitempty"`
+
+	// A list of CIDRs that should be able to reach the opened ports
+	// for an exposed application's endpoint.
+	ExposeToCIDRs []string `bson:"to-cidrs,omitempty"`
+}
+
+// AllowTrafficFromAnyNetwork returns true if the exposed endpoint parameters
+// include the 0.0.0.0/0 CIDR.
+func (exp ExposedEndpoint) AllowTrafficFromAnyNetwork() bool {
+	for _, cidr := range exp.ExposeToCIDRs {
+		if cidr == firewall.AllNetworksIPV4CIDR {
```

Comment:
Shame we don't have a nice abstraction over a predicate here - `if predicate(cidr) {`. Then you can ask better questions about `CIDRs`